pkg/sdk/urls.go

```diff
@@ -1,10 +1,32 @@
 package sdk
 
-import "fmt"
+import (
+	"fmt"
+	"net/url"
+)
 
-// HostToWebsocketURL converts the provided host into a websocket URL that can
+// RPCToWebsocketURL converts the provided URL into a websocket URL string that can
 // be used to subscribe to onchain events and query the chain via a client
 // context or send transactions via a tx client context.
-func HostToWebsocketURL(host string) string {
-	return fmt.Sprintf("ws://%s/websocket", host)
+func RPCToWebsocketURL(hostUrl *url.URL) string {
+	switch hostUrl.Scheme {
+	case "http":
+		return fmt.Sprintf("ws://%s/websocket", hostUrl.Host)
+	case "ws":
+		return fmt.Sprintf("ws://%s/websocket", hostUrl.Host)
+	default:
+		return fmt.Sprintf("wss://%s/websocket", hostUrl.Host)
+	}
+}
+
+// ConstructGRPCUrl constructs a gRPC url string ensuring it contains either the scheme "https" or "http"
```

Comment:
1. The comment says `https` or `http` even though the implementation also included `tcp`
2. The function is called `ContructGRPCURRL` even though it seems to just pass through what it's given?

I'm a bit confused but he intention of this even after reading the comment & function name.

Could you either update or #PUC (Please Update Comment)

Reply:
There were tcp:// schemes in the Makefile and considering that was already a function to help with websocket Urls I figured a similar function for grpc might be useful. My oversight on not updating the comment.

After further consideration and testing we don't need this function so I've removed it.